### PR TITLE
Accept functions

### PR DESCRIPTION
**Pull request by gogonzo**
Part of https://github.com/insightsengineering/teal/issues/1352

Ignores other datasets than `data.frame` and `MultiAssayExperiment`. Unfilterable datasets shouldn't be displayed in the filter-panel as they will be printed independently in the teal-module_data_summary.

### CODE REVIEW

**Review by chlebowa — COMMENTED** [review]
When we made the filter panel "data agnostic" it was considered an overdue success (I cannot find the exact comment that celebrated it).
This change reverses that, the filter panel becomes limited to strictly specified classes again.

File: R/FilteredDataset-utils.R [hunk 1]
```diff
@@ -141,9 +141,5 @@ init_filtered_dataset.default <- function(dataset,
                                           parent, # ignored
                                           join_keys, # ignored
                                           label = attr(dataset, "label", exact = TRUE)) {
-  DefaultFilteredDataset$new(
-    dataset = dataset,
-    dataname = dataname,
-    label = label
-  )
+  NULL
```
Comment (chlebowa) [inline R/FilteredDataset-utils.R:144]: How about an error message?

File: R/FilteredData-utils.R [hunk 2]
```diff
@@ -29,7 +29,10 @@ init_filtered_data <- function(x, join_keys = teal.data::join_keys(), code, chec
       "init_filtered_data(check = 'No longer supported')"
     )
   }
-  FilteredData$new(x, join_keys = join_keys)
+  FilteredData$new(
+    Filter(function(obj) inherits(obj, c("data.frame", "MultiAssayExperiment")), x),
+    join_keys = join_keys
+  )
```
Comment (pawelru) [inline R/FilteredData-utils.R:35]: I echo @chlebowa comment [here](https://github.com/insightsengineering/teal.slice/pull/624#pullrequestreview-2385205495). I think this should be delegated lower into the dataset constructor. Conceptually - a `function` (or anything like that) cannot become a filtered dataset.

**Comment by gogonzo** [comment]
Closing this PR as changes in `teal.slice` are not needed. [PR in teal](https://github.com/insightsengineering/teal/pull/1393) alone fixes the problem.

**Pull request closed without merging**